### DISCUSSION TITLE
UGS will not run after recent Raspbian Bullseye update

### DISCUSSION THREAD

**Discussion by metb40** [winder/Universal-G-Code-Sender#2320] [General]
I had been using bCNC for years with my Raspberry Pi 4 - 4Gb controllers with GRBL 1.1 driver boards on my "shop-built" CNC's. I come from running industrial CNC router for years and am retired now. I have built myself two CNC routers & a 10 watt optical power diode laser machine.
I switched to UGS and loved the way it worked/looked on my 7" touch screens. I prefer this setup due to the dust in the woodshop will ruin a laptop. But after a recent Bullseye update, it no longer will launch. I have tried everything I know of to get my USG to work again, including a complete "clean" install of Raspbian Bullseye  on a new USB 3.0 thumb drive (how I have been running my CNC's for years now... bootable USB), and I can not get it to run again. Being I am a hobby shop and use my CNC's daily, I was forced to go back to bCNC and all machines run fine with it with no issues. The bCNC software is a good software, but the UI is a bit "busy" for my needs. UGS has a much "cleaner/streamlined" UI that serves my needs just fine.
So, does anyone have any ideas on what I could do to get UGS running again? 
Thanks in advance.

**Comment by breiler** [comment]
There has been a couple of bug reports regarding this and I haven't had the time to investigate:
https://github.com/winder/Universal-G-Code-Sender/issues?q=is%3Aissue+is%3Aopen+raspberry+

We probably need to bundle a new Java version with the software in order to fix this. I am currently very busy, so I can't give an estimate time when this will get fixed.

**Reply by metb40** [reply]
Thank you for the information.

**Reply by Egitstuff** [reply]
I tried battling the same issue with the current version of Bullseye. As suggested in one of the discussions, I experimented with editing the conf/ugsplatform.conf script. I eventually modified it to have the complete path after jdkhome = .... This will get UGS to load and most of the functionality seems to be working. However, the gcode editor loads then immediately disappears (I assume it crashes). 

On the other hand, the current nightly build does run with a full install of 32 bit "Buster" on a PI -4 4GB. So it is still quite workable. I assume the release version will run as well, but I need the new record position capability.
